### ISSUE TITLE
Multiple layers of templating aren't resolved

### ISSUE THREAD

**Issue by greg-hellings** [RedHatQE/teflo#50]
If I have multiple layers of Jinja templating, they are not resolved properly. For example, if my SDF file has something like this:

```yaml
name: {{ some_var }}
```

And my vars-data file has something like this:

```yaml
some_var: value-{{ child_var }}
child_var: child_val
```

Then the SDF should resolve to `name: value-child_val` but, instead, it only resolves once to `name: value-{{ child_var }}`. Ansible deals with this by essentially passing the values through the template engine repeatedly until the value does not change between runs. It's actually slightly more complicated than that, because they can handle {% raw %} ... {% endraw %} values. But, for getting part of the way there, just iterating through the engine until the values don't change between consecutive runs would provide enough functionality.

**Comment by rujutashinde** [comment]
That is correct, teflo as of now does not consider multiple layers of jinja templating. I will add a ticket for this

**Comment by JunqiZhang0** [comment]
@greg-hellings the issue should be resolved by #76 , you can take a look

**Comment by rujutashinde** [comment]
Closing this issue Fixed by #76
